### PR TITLE
api: add prochaos; kill some processes in some nodes once or in period

### PR DESCRIPTION
**Pull request by Hexilee**
Signed-off-by: hexilee <i@hexilee.me>

### What problem does this PR solve?

Problem Summary:

We need process chaos in some test cases; this or support kill some processes by pattern once or in period.

### What is changed and how it works?

How it Works:

Add CR `ProcChaos` and its reconciler. Bind the procchaos with a request and apply it by the reconciler.

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

Deploy a cluster, then modify `pipeline-examples/proc-chaos.yaml` and apply it.

### CODE REVIEW

File: controllers/procchaos_controller.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+			}
+		}
+	}
+
+	if !nodeFound {
+		err = fmt.Errorf("node(%s) is not exist", node)
+		return
+	}
+
+	if resource.Status.State != naglfarv1.MachineReady {
```
Comment (mahjonp) [inline controllers/procchaos_controller.go:158]:
```suggestion
	if resource.Status.State != naglfarv1.ResourceReady {
```

**Review by mahjonp — APPROVED** [review]
Rest LGTM

File: controllers/procchaos_controller.go (hunk 1 above)
Comment (Hexilee) [reply controllers/procchaos_controller.go]: Fixed

**Review by mahjonp — APPROVED** [review]
LGTM

**Pull request merged**